pkg/client/client.go

```diff
@@ -517,6 +521,46 @@ func (w *uploadChunkWorker) Start() <-chan error {
 }
 
 func (w *uploadChunkWorker) upload(job uploadChunkJob) error {
+	// return errors.New("abc")
+	pipeReader, pipeWriter := io.Pipe()
+	writer := multipart.NewWriter(pipeWriter)
+	go func() {
+		defer pipeWriter.Close()
+		defer writer.Close()
+		if err := writeMultipartRequest(writer, job); err != nil {
+			fmt.Fprintf(w.DumpOut, "Error writing multipart request %v", err)
```

Comment:
call the context's cancel on error copying the file.

That should cause the request to be cancelled instead of sent with incomplete/invalid data and wait for the server's error response.

Reply:
good idea, used a new context for each request though, this way different requests are not tied to the same context.

Reply:
it's ok to use a per-request context, but the parent of those contexts needs to be the worker's context, not `context.Background()`.

Reply:
agree, done